### PR TITLE
ESP32P: Add CI for ESP32H2

### PR DESCRIPTION
**Pull request by wqx6**
#### Problem
Add CI for ESP32H2

#### Change overview
* Add CI for ESP32H2
* move `chip_enable_wifi` from `/src/platform/device.gni` to `config/esp32/components/chip/CMakeLists.txt` for esp32 platform

#### Testing
Passed ESP32-H2 CI

### CODE REVIEW

File: scripts/examples/esp_example.sh [hunk 1]
```diff
@@ -30,20 +30,40 @@ if [ -z "$app" ]; then
     exit 1
 fi
 
-source "$IDF_PATH/export.sh"
+if [ "$sdkconfig_name" == "sdkconfig.defaults.esp32h2" ]; then
+    pushd .
+    cd "$IDF_PATH"
+    git fetch origin master
+    git checkout 10f3aba770
```
Comment (andy31415) [inline scripts/examples/esp_example.sh:37]: This seems off - IDF should be configured in advance. This checks out a specific version and it will not revert it.

I believe build scripts should not touch SDKs but rather  check that SDKs are correctpy configured.
We need an image that has the right version.

File: src/platform/device.gni [hunk 2]
```diff
@@ -55,10 +55,10 @@ declare_args() {
 
   # Enable wifi support.
   chip_enable_wifi =
-      chip_device_platform == "linux" || chip_device_platform == "esp32" ||
-      chip_device_platform == "mbed" || chip_device_platform == "tizen" ||
-      chip_device_platform == "android" || chip_device_platform == "ameba" ||
-      chip_device_platform == "webos" || chip_device_platform == "bl602"
+      chip_device_platform == "linux" || chip_device_platform == "mbed" ||
```
Comment (andy31415) [inline src/platform/device.gni:58]: Is this change needed? the enable_wifi seems to be a default, so "on" or "off" for a platform that supports both does not seem to be as relevant. Not having this change seems to result in a smaller delta.

File: scripts/examples/esp_example.sh (hunk 1 above)
Comment (andy31415) [reply scripts/examples/esp_example.sh:37]: The reason I care about this is that developers use a single image to compile things. I want to be able to compile anything without making my SDK/tree dirty. I should be able to build H2 as well as others after each other in the same environment.

CI will start fresh every time, so we do not care as much, however developer images (usually the chip-build-vscode and similar variants) are persistent.

**Review by andy31415 — CHANGES_REQUESTED** [review]
Requesting changes: build scripts should not have sideffects outside of build output (i.e. they should not dirty tree, change SDKs or install additional packages)

File: src/platform/device.gni (hunk 2 above)
Comment (wqx6) [reply src/platform/device.gni]: removed this change.

File: scripts/examples/esp_example.sh (hunk 1 above)
Comment (andy31415) [reply scripts/examples/esp_example.sh:37]: We could add a separate docker image in that case or make this update part of CI explicitly rather than build scripts. Build scripts should not touch SDK/source in a such strong way.
